server/broadcast.go

```diff
@@ -1571,6 +1571,12 @@ func updateSession(sess *BroadcastSession, res *ReceivedTranscodeResult) {
 	}
 }
 
+func clearSessionBalance(sess *BroadcastSession, id core.ManifestID) {
+	sess.lock.Lock()
+	defer sess.lock.Unlock()
+	sess.Balance = core.NewBalance(ethcommon.BytesToAddress(sess.OrchestratorInfo.TicketParams.Recipient), id, sess.Balances)
```

Comment:
Sometimes it still panics on this line - not always though

I run offchain so not sure why this is being triggered, maybe we need an offchain check somewhere

backtrace below

```
josh@josh-gpu:~/go-livepeer$ ./livepeer -gateway -rtmpAddr :1936 -httpAddr :5936 -orchAddr localhost:8935 2>&1 | tee gw.log
*----------*----------------*
| RtmpAddr | :1936          |
| HttpAddr | :5936          |
| OrchAddr | localhost:8935 |
| Gateway  | true           |
*----------*----------------*
I1209 21:14:26.986319 2183666 starter.go:470] ***Livepeer is running on the offchain network***
I1209 21:14:26.987591 2183666 starter.go:638] ***Livepeer is in off-chain mode***
I1209 21:14:26.997546 2183666 starter.go:1666] ***Livepeer Running in Gateway Mode***
I1209 21:14:26.997827 2183666 starter.go:1667] Video Ingest Endpoint - rtmp://127.0.0.1:1936
I1209 21:14:26.998029 2183666 starter.go:1676] Livepeer Node version: 0.8.1-ba63fe46-dirty
I1209 21:14:26.998171 2183666 webserver.go:20] CLI server listening on 127.0.0.1:5935
I1209 21:14:40.609518 2183666 ai_mediaserver.go:468] stream=cosmos source_id=d94baab9-0d7a-492e-aeb2-f8a26e8da08d source_type=rtmpConn remote_addr=127.0.0.1 request_id=7bc0432f Received live video AI requ
est for cosmos
I1209 21:14:40.615945 2183666 discovery.go:180] Done fetching orch info numOrch=1 responses=1/1 timedOut=false
I1209 21:14:40.616047 2183666 panic.go:785] remote_addr=127.0.0.1 request_id=7bc0432f capability=Live video to video stream=cosmos source_id=d94baab9-0d7a-492e-aeb2-f8a26e8da08d source_type=rtmpConn Processed AI request model_id=noop took=250ns
2024/12/09 21:14:40 http: panic serving 127.0.0.1:39034: runtime error: invalid memory address or nil pointer dereference
goroutine 24 [running]:
net/http.(*conn).serve.func1()
        /home/josh/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.23.2.linux-amd64/src/net/http/server.go:1947 +0xbe
panic({0x186ac60?, 0x2baff50?})
        /home/josh/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.23.2.linux-amd64/src/runtime/panic.go:785 +0x132
github.com/livepeer/go-livepeer/server.clearSessionBalance(0xc0008b8600, {0xc0005f0848, 0x8})
        /home/josh/go-livepeer/server/broadcast.go:1577 +0x70
github.com/livepeer/go-livepeer/server.submitLiveVideoToVideo({0x1edbf58, 0xc00073d230}, {0xc0001bd000, {0x1ee7e70, 0xc0009541c0}, 0xc0004fc260, {0xc0005ee3c0, {0xc000942140, 0x1b}, {0xc000930a6a, ...}, ...}}, ...)
        /home/josh/go-livepeer/server/ai_process.go:1015 +0x86
github.com/livepeer/go-livepeer/server.processAIRequest.func10({0x1edbf58?, 0xc00073d230?}, {0xc0001bd000, {0x1ee7e70, 0xc0009541c0}, 0xc0004fc260, {0xc0005ee3c0, {0xc000942140, 0x1b}, {0xc000930a6a, ...}, ...}}, ...)
        /home/josh/go-livepeer/server/ai_process.go:1422 +0x7f
github.com/livepeer/go-livepeer/server.processAIRequest({0x1edbf58?, 0xc00073d230?}, {0xc0001bd000, {0x1ee7e70, 0xc0009541c0}, 0xc0004fc260, {0xc0005ee3c0, {0xc000942140, 0x1b}, {0xc000930a6a, ...}, ...}}, ...)  
        /home/josh/go-livepeer/server/ai_process.go:1462 +0xdf5
github.com/livepeer/go-livepeer/server.startAIMediaServer.(*LivepeerServer).StartLiveVideo.func12({0x1ed7680, 0xc0008b61c0}, 0xc0006a6140)
        /home/josh/go-livepeer/server/ai_mediaserver.go:501 +0xa74
net/http.HandlerFunc.ServeHTTP(0xc0008b6000?, {0x1ed7680?, 0xc0008b61c0?}, 0x2?)
        /home/josh/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.23.2.linux-amd64/src/net/http/server.go:2220 +0x29
net/http.(*ServeMux).ServeHTTP(0x416805?, {0x1ed7680, 0xc0008b61c0}, 0xc0006a6140)
        /home/josh/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.23.2.linux-amd64/src/net/http/server.go:2747 +0x1ca
net/http.serverHandler.ServeHTTP({0x1ed07e0?}, {0x1ed7680?, 0xc0008b61c0?}, 0x6?)
        /home/josh/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.23.2.linux-amd64/src/net/http/server.go:3210 +0x8e
net/http.(*conn).serve(0xc000956000, {0x1edbf58, 0xc0002f8ff0})
        /home/josh/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.23.2.linux-amd64/src/net/http/server.go:2092 +0x5d0
created by net/http.(*Server).Serve in goroutine 36
        /home/josh/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.23.2.linux-amd64/src/net/http/server.go:3360 +0x485
```

Reply:
Ahh, yeah, sorry, my bad, for offchain, there are no balances.

Here's the fix: https://github.com/livepeer/go-livepeer/pull/3306